Framework/Core/test/test_DataDescriptorMatcherWorkflow.cxx

```diff
@@ -0,0 +1,150 @@
+
+
+#define ASSERT_ERROR(condition)                                   \
+  if ((condition) == false) {                                     \
+    LOG(fatal) << R"(Test condition ")" #condition R"(" failed)"; \
+  }
+
+using namespace o2::framework;
+using DataHeader = o2::header::DataHeader;
+
+std::vector<DataProcessorSpec> defineDataProcessing(ConfigContext const& config)
+{
+  std::vector<DataProcessorSpec> workflow;
+
+  auto producerCallback = [counter = std::make_shared<size_t>()](DataAllocator& outputs, ControlService& control) {
+    if (*counter > 0) {
+      // don't know if we enter the processing callback after the EOS was sent
+      return;
+    }
+    outputs.make<unsigned int>(Output{"TST", "SAMPLE", 1}) = 1;
+    outputs.make<unsigned int>(Output{"TST", "SAMPLE", 2}) = 2;
+    ++(*counter);
+    control.endOfStream();
+  };
+
+  workflow.emplace_back(DataProcessorSpec{"producer",
+                                          {InputSpec{"timer", "TST", "TIMER", 0, Lifetime::Timer}},
+                                          {OutputSpec{"TST", "SAMPLE", 1, Lifetime::Timeframe},
+                                           OutputSpec{"TST", "SAMPLE", 2, Lifetime::Timeframe}},
+                                          AlgorithmSpec{adaptStateless(producerCallback)},
+                                          {ConfigParamSpec{"period-timer", VariantType::Int, 100000, {"timer"}}}});
+
+  auto processorCallback = [counter = std::make_shared<size_t>()](InputRecord& inputs, DataAllocator& outputs) {
+    // should not be called more than one time
+    ASSERT_ERROR(*counter == 0);
+    ++(*counter);
+    int nBlocks = 0;
+    for (auto ref : InputRecordWalker(inputs)) {
+      auto const* dh = DataRefUtils::getHeader<DataHeader*>(ref);
+      ASSERT_ERROR(dh != nullptr);
+      auto const& data = inputs.get<unsigned int>(ref);
+      ASSERT_ERROR(data == dh->subSpecification);
+      outputs.make<unsigned int>(OutputRef{"out", 0}) = data;
+      LOG(info) << fmt::format("forwarded {}/{}/{} with data {}",
+                               dh->dataOrigin.as<std::string>(),
+                               dh->dataDescription.as<std::string>(),
+                               dh->subSpecification,
+                               data);
+      ++nBlocks;
+    }
+    ASSERT_ERROR(nBlocks == 2);
+  };
+
+  using DataDescriptorMatcher = o2::framework::data_matcher::DataDescriptorMatcher;
+  DataDescriptorMatcher processorInputMatcher = {
+    DataDescriptorMatcher::Op::And,
+    data_matcher::OriginValueMatcher{"TST"},
+    std::make_unique<DataDescriptorMatcher>(
+      DataDescriptorMatcher::Op::And,
+      data_matcher::DescriptionValueMatcher{"SAMPLE"},
+      std::make_unique<DataDescriptorMatcher>(
+        DataDescriptorMatcher::Op::And,
+        std::make_unique<DataDescriptorMatcher>(
+          DataDescriptorMatcher::Op::Xor,
+          data_matcher::SubSpecificationTypeValueMatcher{0},
+          data_matcher::ConstantValueMatcher{true}),
```

Comment:
Since it's two bool variables which will be logically xor'ed, this corresponds to a negation.

It's a demonstrator and it's working. I will implement the `DataDescriptorMatcher::Op::Not` alongside `Op::Just`, this will probably  not require the other change above, but the 'Not' operator might have similar implications on the uniqueness of the matcher parameters.